docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Testing Tricky Edge Cases in a TUF Client"
+author: Ivana Atanasova
+---
+
+Usually the TUF Specification creates an impression of simple and straightforward approach to address software update systems security gaps. In the next few paragraphs we'll try to convince you that the devil is in the details.
+
+With the [v1.0.0 release](https://blogs.vmware.com/opensource/2022/02/22/python-tuf-reaches-version-1-0-0/) we can say that the current reference implementation is finally in a good place, although it wouldn’t be so trustworthy without all the awesome test functionality it provides. Therein lies some interesting surprises, for the test coverage reflects use cases that wouldn’t respectively come to mind. TUF, in fact, is capable of managing some tricky business! And that’s why it provides more than just test coverage - it is a way to analyze and validate that the implementation strictly follows the TUF spec with all its corner cases and tricky details that are far from obvious.
+Before looking into them, we need to introduce the test functionality itself.
+
+
+## Some Repository Simulator Magic
+
+The great thing about [RepositorySimulator](https://github.com/theupdateframework/python-tuf/blob/develop/tests/repository_simulator.py), is it allows you to play with repository metadata by modifying it, signing and storing new roles versions, while serving older ones in the client test code. You can also simulate downloading new metadata from a remote without the need of file access or network connections, and modify expiry dates and time.
+
+When you execute a test in python-tuf, it creates a local temporary directory serving for cache, with "/metadata/..." and "/targets/..." URL paths that host metadata and targets respectively.
```

Comment:
oh yeah this is actually incorrect: RepositorySimulator does not create directories, it serves everything from memory

Reply:
This could actually make for a good transition to advertising the `--dump` flag. Something along the lines of, "Although, RepositorySimulator hosts repos purely in memory, you can supply the `--dump` flag to write its contents to a temporary directory on the local filesystem in order to audit the metadata."  (feel free to rephrase as you see fit)